src/shark/service_token/node_client.js

```diff
@@ -0,0 +1,87 @@
+'use strict'
+
+import { Deserializer } from 'jsonapi-serializer'
+import { isString } from 'src/utils/typecheck'
+import signedFetch from 'src/utils/signed_fetch'
+
+const deserializer = new Deserializer({ keyForAttribute: 'camelCase' })
+
+/**
+ * @class ServiceTokenClient
+ * @classdesc Helper class to request and manage a valid service token in a NodeJS environment.
+ */
+export default class ServiceTokenClient {
+  /**
+   * Remove stored service token
+   */
+  static reset () {
+  }
+
+  constructor (options = {}) {
+    this.accessKey = options.accessKey
+    this.secretKey = options.secretKey
+    this.digest = options.digest || 'sha1'
+    this.baseUrl = options.baseUrl
+
+    if (!isString(this.baseUrl)) {
+      throw new Error('Parameter `baseUrl` is missing or not a string')
+    }
+    if (!isString(this.accessKey)) {
+      throw new Error('Parameter `accessKey` is missing or not a string')
+    }
+    if (!isString(this.secretKey)) {
+      throw new Error('Parameter `secretKey` is missing or not a string')
+    }
+  }
+
+  /**
+   * Create a new service token.
+   *
+   * @return {Promise} the fetch promise
+   */
+  createServiceToken (params, options = {}) {
+    const url = `${this.baseUrl}/api/tokens/service_token`
+    const { userId, customClaims } = params
+    const requestOptions = Object.assign({
+      method: 'POST',
+      body: {
+        user_id: userId,
+        custom_claims: customClaims || {}
+      }
+    }, options)
+
+    return this.__request(url, requestOptions)
+  }
+
+  /**
+   * Verifies if a service token is still valid.
+   *
+   * @return {Promise} the fetch promise
+   */
+  verifyServiceToken (params, options = {}) {
+    const url = `${this.baseUrl}/api/users/authenticate?service_token=${params.serviceToken}`
```

Comment:
That weird construct with the `urls` object containing functions to return URLs for each function in my original implementation was so you could override the actual URL more easily. Whilst I guess _that_ much "power" might have been a bit too much, I think being able to specify additional query string parameters such as `include` might be relevant at some point. Especially if we need do include `permissions` in some, but not all objects.

Alternatively, `verifyServiceToken` could _always_ have `include=permissions` in the URL. Given that some actions like creating notifications also care about authorisation, that's necessary, unless we allow for a way to modify the endpoint URL.

Reply:
Actually it should be in the params:
`params = { serviceToken: '0123456789', include: 'permissions' }`. 

I just forgot to add the params to the query string.